src/System.Text.Json/src/System/Text/Json/JsonException.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Defines a custom exception object that is thrown when invalid JSON text is encountered, when the defined maximum depth is passed,
+    /// or the JSON text is not compatible with the type of a property on an object.
+    /// </summary>
+    [Serializable]
+    public class JsonException : Exception
+    {
+        /// <summary>
+        /// Creates a new exception object to relay error information to the user.
+        /// </summary>
+        /// <param name="message">The context specific error message.</param>
+        /// <param name="lineNumber">The line number at which the invalid JSON was encountered (starting at 0) when deserializing.</param>
+        /// <param name="bytePositionInLine">The byte count within the current line where the invalid JSON was encountered (starting at 0).</param>
+        /// <param name="path">The object's property path where the invalid JSON was encountered.</param>
+        /// <param name="innerException">The exception that caused the current exception.</param>
+        /// <remarks>
+        /// Note that the <paramref name="bytePositionInLine"/> counts the number of bytes (i.e. UTF-8 code units) and not characters or scalars.
+        /// </remarks>
+        public JsonException(string message, string path, long? lineNumber, long? bytePositionInLine, Exception innerException) : base(message, innerException)
```

Comment:
We should add a remark that the line number/position won't be set by writer/serialization, hence null.